backend/src/test/java/org/uresti/pozarreal/service/impl/CourseAssistantPaymentServiceImplTests.java

```diff
@@ -0,0 +1,219 @@
+                .thenReturn(coursePaymentList);
+
+        // When:
+        List<CoursePayment> courseAssistant = courseAssistantPaymentService
+                .findAllByCourseAssistant("abc");
+
+        // Then:
+        Assertions.assertThat(courseAssistant.size()).isEqualTo(2);
+        Assertions.assertThat(courseAssistant.get(0).getId()).isEqualTo("id1");
+        Assertions.assertThat(courseAssistant.get(0).getPaymentDate()).isEqualTo(now);
+        Assertions.assertThat(courseAssistant.get(0).getUserId()).isEqualTo("User 1");
+        Assertions.assertThat(courseAssistant.get(0).getAmount()).isEqualTo(new BigDecimal(50));
+        Assertions.assertThat(courseAssistant.get(0).getCourseAssistantId()).isEqualTo("abc");
+        Assertions.assertThat(courseAssistant.get(0).getReceiptNumber()).isEqualTo("12345");
+        Assertions.assertThat(courseAssistant.get(0).getNotes()).isEqualTo("hello");
+        Assertions.assertThat(courseAssistant.get(0).getConcept()).isEqualTo("Concept 1");
+        Assertions.assertThat(courseAssistant.get(1).getId()).isEqualTo("id2");
+        Assertions.assertThat(courseAssistant.get(1).getPaymentDate()).isEqualTo(now);
+        Assertions.assertThat(courseAssistant.get(1).getUserId()).isEqualTo("User 2");
+        Assertions.assertThat(courseAssistant.get(1).getAmount()).isEqualTo(new BigDecimal(50));
+        Assertions.assertThat(courseAssistant.get(1).getCourseAssistantId()).isEqualTo("abc");
+        Assertions.assertThat(courseAssistant.get(1).getReceiptNumber()).isEqualTo("12345");
+        Assertions.assertThat(courseAssistant.get(1).getNotes()).isEqualTo("hello");
+        Assertions.assertThat(courseAssistant.get(1).getConcept()).isEqualTo("Concept 2");
+    }
+
+    @Test
+    public void givenAnCoursePayment_whenSaved_thenIsSaved() {
+        // Given:
+        CourseAssistantPaymentRepository courseAssistantPaymentRepository = Mockito
+                .mock(CourseAssistantPaymentRepository.class);
+
+        CourseAssistantPaymentServiceImpl courseAssistantPaymentService = new
+                CourseAssistantPaymentServiceImpl(courseAssistantPaymentRepository);
+
+        LocalDate now = LocalDate.now();
+
+        org.uresti.pozarreal.model.CoursePayment coursePayment = org.uresti.pozarreal.model.CoursePayment.builder()
+                .id("id1")
+                .paymentDate(now)
+                .amount(new BigDecimal(50))
+                .courseAssistantId("abc")
+                .receiptNumber("12345")
+                .notes("hello")
+                .userId("User 1")
+                .concept("Concept 1")
+                .build();
+
+        Mockito.when(courseAssistantPaymentRepository.save(coursePayment)).thenReturn(coursePayment);
+
+        // When:
+        CoursePayment courseAssistant = courseAssistantPaymentService
+                .save(CourseAssistantPaymentMapper.entityToDto(coursePayment));
```

Comment:
Don't use CourseAssistantPaymentMapper.entityToDto(coursePayment) calling the save method that you're testing ... pass only the object that you expect to receive. Build the object manually (I mean without entityToDto)

Reply:
done